toolkit/tools/pkg/imagecustomizerlib/customizegroups.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/userutils"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+func AddOrUpdateGroups(ctx context.Context, groups []imagecustomizerapi.Group,
+	imageChroot safechroot.ChrootInterface,
+) error {
+	if len(groups) == 0 {
+		return nil
+	}
+
+	_, span := otel.GetTracerProvider().Tracer(OtelTracerName).Start(ctx, "add_or_update_groups")
+	span.SetAttributes(
+		attribute.Int("groups_count", len(groups)),
+	)
+	defer span.End()
+
+	for _, group := range groups {
+		err := addOrUpdateGroup(group, imageChroot)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func addOrUpdateGroup(group imagecustomizerapi.Group, imageChroot safechroot.ChrootInterface,
+) error {
+	// Check if the user already exists.
+	groupExists, err := userutils.GroupExists(group.Name, imageChroot)
+	if err != nil {
+		return err
+	}
+
+	if groupExists {
```

Comment:
nit: merge this if with the next?

Reply:
Fixed.

(I was following the pattern from users, which is more complex.)